ProteGO/Common/BeaconIdsManager/BeaconIdsManager.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import RealmSwift
+
+final class BeaconIdsManager: BeaconIdsManagerType {
+
+    var allBeaconIds: Results<RealmExpiringBeacon> {
+        return self.realmManager.realm.objects(RealmExpiringBeacon.self).sorted(byKeyPath: "expirationDate", ascending: true)
+    }
+
+    var lastStoredExpiringBeaconDate: Date? {
+        return self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .sorted(byKeyPath: "expirationDate", ascending: false)
+            .first?
+            .expirationDate
+    }
+
+    var currentExpiringBeaconId: ExpiringBeaconId? {
+        let currentDate = currentDateProvider.currentDate
+
+        guard let previousExpiringBeacon = self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .filter("expirationDate <= %@", currentDate).sorted(byKeyPath: "expirationDate", ascending: false).first else {
+                // no valid beacon found
+                return nil
+        }
+
+        guard let previousBeacon = BeaconId(data: previousExpiringBeacon.beaconIdData) else {
+            logger.error("Incorrect beacon found \(previousExpiringBeacon.beaconIdData.toHexString())")
+            return nil
+        }
+
+        guard let nextExpiringBeacon = self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .filter("expirationDate > %@", currentDate).sorted(byKeyPath: "expirationDate", ascending: true).first else {
+                // last beacon with current date + default lifespan
+                let date = currentDate.addingTimeInterval(Constants.Bluetooth.ExpiringBeaconDefaultLifespan)
```

Comment:
Czy tutaj nie powinniśmy dodać default lifespan do `previousExpiringBeacon.expirationDate` zamiast do obecnej daty?

Reply:
No właśnie nie, bo jezeli jest to ostatni beacon, i nie mamy nic nowszego to ten beacon byłby przeterminowany juz na starcie.

Ten przypadek to jest po prostu powtarzanie ostatniego beacona w nieskonczoność

Reply:
Ok, racja 👍